scripts/npc/1012000.js

```diff
@@ -0,0 +1,53 @@
+// Henesys Regular Cab
+
+var towns = [104000000, 102000000, 101000000, 103000000]
+var prices_text = ["800", "1,000", "1,000", "1,200"]
+var prices_num = [800, 1000, 1000, 1200]
+
+var state = 0
+var selection = -1
```

Comment:
Unfortunately the VM doesn't support `const` and `let`.

Each script has it's own global namespace so cross-script pollution should not occur.

We can't define state within the `run` function as run gets executed for each stage of the script, meaning state would be reset upon each execution (I might have misunderstood what you are describing here, if so an example would be appreciated as I am not that familiar with js).

The scripts are run in the following manner:

1. Compile to byte code (this will generate and populate the global namespace)
2. call `run`
4. Read script action and set any flags e.g. yes, no, next, back, selection
5. goto 2

Reply:
o wow, i see; ok the current approach makes sense if that is the case!